default/python/AI/ProductionAI.py

```diff
@@ -1081,9 +1081,15 @@ def generate_production_orders():
         else:
             warn("Failed enqueueing %s at planet %s, got result %d" % (building_name, planet, res))
 
-    queued_clny_bld_locs = [element.locationID for element in production_queue if element.name.startswith('BLD_COL_')]
-    colony_bldg_entries = ([entry for entry in foAI.foAIstate.colonisablePlanetIDs.items() if entry[1][0] > 60 and
-                           entry[0] not in queued_clny_bld_locs and entry[0] in state.get_empire_outposts()]
+    # ignore acquired-under-construction colony buildings for which our empire lacks the species
+    queued_clny_bld_locs = [element.locationID for element in production_queue
+                            if (element.name.startswith('BLD_COL_') and
+                                empire_has_colony_bld_species(element.name))]
+    colony_bldg_entries = ([entry for entry in foAI.foAIstate.colonisablePlanetIDs.items() if
+                                entry[1][0] > 60 and
+                                entry[0] not in queued_clny_bld_locs and
+                                entry[0] in state.get_empire_outposts() and
+                                not already_has_completed_colony_building(entry[0])]
```

Comment:
Precedes this PR but should probably unpack the entry tuple into descriptive names rather than using cryptic indexing...